python/lsst/ip/diffim/dcrModel.py

```diff
@@ -0,0 +1,619 @@
+
+    Attributes
+    ----------
+    dcrNumSubfilters : `int`
+        Number of sub-filters used to model chromatic effects within a band.
+    filterInfo : `lsst.afw.image.Filter`
+        The filter definition, set in the current instruments' obs package.
+    modelImages : `list` of `lsst.afw.image.MaskedImage`
+        A list of masked images, each containing the model for one subfilter
+
+    Parameters
+    ----------
+    modelImages : `list` of `lsst.afw.image.MaskedImage`
+        A list of masked images, each containing the model for one subfilter.
+    filterInfo : `lsst.afw.image.Filter`, optional
+        The filter definition, set in the current instruments' obs package.
+        Required for any calculation of DCR, including making matched templates.
+
+    Notes
+    -----
+    The ``DcrModel`` contains an estimate of the true sky, at a higher
+    wavelength resolution than the input observations. It can be forward-
+    modeled to produce Differential Chromatic Refraction (DCR) matched
+    templates for a given ``Exposure``, and provides utilities for conditioning
+    the model in ``dcrAssembleCoadd`` to avoid oscillating solutions between
+    iterations of forward modeling or between the subfilters of the model.
+    """
+
+    def __init__(self, modelImages, filterInfo=None, psf=None):
+        self.dcrNumSubfilters = len(modelImages)
+        self.modelImages = modelImages
+        self._filter = filterInfo
+        self._psf = psf
+
+    @classmethod
+    def fromImage(cls, maskedImage, dcrNumSubfilters, filterInfo=None, psf=None):
+        """Initialize a DcrModel by dividing a coadd between the subfilters.
+
+        Parameters
+        ----------
+        maskedImage : `lsst.afw.image.MaskedImage`
+            Input coadded image to divide equally between the subfilters.
+        dcrNumSubfilters : `int`
+            Number of sub-filters used to model chromatic effects within a band.
+        filterInfo : `lsst.afw.image.Filter`, optional
+            The filter definition, set in the current instruments' obs package.
+            Required for any calculation of DCR, including making matched templates.
+        psf : `lsst.afw.detection.Psf`, optional
+            Point spread function (PSF) of the model.
+            Required if the ``DcrModel`` will be persisted.
+
+        Returns
+        -------
+        dcrModel : `lsst.pipe.tasks.DcrModel`
+            Best fit model of the true sky after correcting chromatic effects.
+        """
+        # NANs will potentially contaminate the entire image,
+        # depending on the shift or convolution type used.
+        model = maskedImage.clone()
+        badPixels = np.isnan(model.image.array) | np.isnan(model.variance.array)
+        model.image.array[badPixels] = 0.
+        model.variance.array[badPixels] = 0.
+        model.image.array /= dcrNumSubfilters
+        model.variance.array /= dcrNumSubfilters
```

Comment:
I believe so, yes. Since the subfilter images are treated as independent, when we divide an image into N subfilters I think each should be treated as having less information, and sqrt(N) worse signal to noise. I will add unit tests in DM-15342 and DM-15343 that will test this and hopefull clarify the expected behavior.